File: src/main/java/com/github/novicezk/midjourney/enums/MessageType.java
package com.github.novicezk.midjourney.enums;


public enum MessageType {
	/**
	 * Create.
	 */
	CREATE,
	/**
	 * Update.
	 */
	UPDATE,
	/**
	 * Delete.
	 */
	DELETE;

	public static MessageType of(String type) {
		return switch (type) {
			case "MESSAGE_CREATE" -> CREATE;
			case "MESSAGE_UPDATE" -> UPDATE;
			case "MESSAGE_DELETE" -> DELETE;
			default -> null;
		};
	}
}

File: src/enums/MessageType.ts
/**
 * Discord message types
 */
export enum MessageType {
  /** Create */
  CREATE = 'CREATE',
  /** Update */
  UPDATE = 'UPDATE',
  /** Delete */
  DELETE = 'DELETE',
}

/**
 * Get message type from Discord event type string
 */
export function getMessageType(type: string): MessageType | null {
  switch (type) {
    case 'MESSAGE_CREATE':
      return MessageType.CREATE;
    case 'MESSAGE_UPDATE':
      return MessageType.UPDATE;
    case 'MESSAGE_DELETE':
      return MessageType.DELETE;
    default:
      return null;
  }
}

